apps/multiplatform/common/src/commonMain/kotlin/chat/simplex/common/views/contacts/ContactListNavView.kt

```diff
@@ -0,0 +1,112 @@
+import chat.simplex.common.views.helpers.*
+import chat.simplex.res.MR
+import kotlinx.coroutines.delay
+
+@Composable
+fun ContactListNavLinkView(chat: Chat, nextChatSelected: State<Boolean>) {
+    val showMenu = remember { mutableStateOf(false) }
+    val disabled = chatModel.chatRunning.value == false || chatModel.deletedChats.value.contains(chat.remoteHostId to chat.chatInfo.id)
+    LaunchedEffect(chat.id) {
+        showMenu.value = false
+        delay(500L)
+    }
+    val selectedChat = remember(chat.id) { derivedStateOf { chat.id == chatModel.chatId.value } }
+
+    when (chat.chatInfo) {
+        is ChatInfo.Direct -> {
+            ChatListNavLinkLayout(
+                chatLinkPreview = {
+                    tryOrShowError("${chat.id}ContactListNavLink", error = { ErrorChatListItem() }) {
+                        ContactPreviewView(chat, disabled)
+                    }
+                },
+                click = {
+                    directChatAction(chat.remoteHostId, chat.chatInfo.contact, chatModel)
+                    ModalManager.start.closeModals()
+                },
+                dropdownMenuItems = {
+                    tryOrShowError("${chat.id}ContactListNavLinkDropdown", error = {}) {
+                        ContactMenuItems(chat, chat.chatInfo.contact, chatModel, showMenu)
+                    }
+                },
+                showMenu,
+                disabled,
+                selectedChat,
+                nextChatSelected,
+            )
+        }
+        is ChatInfo.ContactRequest -> {
+            ChatListNavLinkLayout(
+                chatLinkPreview = {
+                    tryOrShowError("${chat.id}ContactListNavLink", error = { ErrorChatListItem() }) {
+                        ContactPreviewView(chat, disabled)
+                    }
+                },
+                click = {
+                    contactRequestAlertDialog(
+                        chat.remoteHostId,
+                        chat.chatInfo,
+                        chatModel,
+                        onSuccess = {
+                            ModalManager.start.closeModals()
+                        }
+                    )
+                },
+                dropdownMenuItems = {
+                    tryOrShowError("${chat.id}ContactListNavLinkDropdown", error = {}) {
+                        ContactRequestMenuItems(chat.remoteHostId, chat.chatInfo, chatModel, showMenu)
+                    }
+                },
+                showMenu,
+                disabled,
+                selectedChat,
+                nextChatSelected,
+            )
+        }
+        else -> {}
+    }
+}
+
+@Composable
+fun ContactMenuItems(chat: Chat, contact: Contact, chatModel: ChatModel, showMenu: MutableState<Boolean>) {
+    if (contact.activeConn != null) {
+        ToggleFavoritesChatAction(chat, chatModel, chat.chatInfo.chatSettings?.favorite == true, showMenu)
+    }
+    DeleteContactAction(chat, chatModel, showMenu)
```

Comment:
I think DeleteContactAction shouldn't be available on this view - it's not a "new chat" + it's not necessary to find a place to delete only contact w/t deleting conversation now as all options will be available from both chat list and chat info

Reply:
favorite is probably ok to keep as it affects toggle above

Reply:
as discussed, let's remove all context menus in this list